components/Modal/Restake.vue

```diff
@@ -0,0 +1,232 @@
+        You don't have any tokens staked with this validator.
+      </span>
+      <FormMessage
+        v-else-if="$v.amount.$error && (!$v.amount.required || amount === 0)"
+        name="Amount"
+        type="required"
+      />
+      <FormMessage
+        v-else-if="$v.amount.$error && !$v.amount.decimal"
+        name="Amount"
+        type="numeric"
+      />
+      <FormMessage
+        v-else-if="$v.amount.$error && !$v.amount.max"
+        type="custom"
+        :msg="`You don't have enough ${stakingDenom} to proceed.`"
+      />
+      <FormMessage
+        v-else-if="$v.amount.$error && !$v.amount.min"
+        :min="smallestAmount"
+        name="Amount"
+        type="min"
+      />
+      <FormMessage
+        v-else-if="$v.amount.$error && !$v.amount.maxDecimals"
+        name="Amount"
+        type="maxDecimals"
+      />
+    </CommonFormGroup>
+  </ModalAction>
+</template>
+
+<script>
+import { mapState } from 'vuex'
+import { required, decimal } from 'vuelidate/lib/validators'
+import { orderBy } from 'lodash'
+import { SMALLEST } from '~/common/numbers'
+import { validatorEntry } from '~/common/address'
+import { lunieMessageTypes } from '~/common/lunie-message-types'
+import network from '~/common/network'
+
+export default {
+  name: `RestakeModal`,
+  filters: {
+    validatorEntry,
+  },
+  props: {
+    sourceValidator: {
+      type: Object,
+      default: () => ({}),
+    },
+    validators: {
+      type: Array,
+      required: true,
+    },
+  },
+  data: () => ({
+    amount: null,
+    lunieMessageTypes,
+    smallestAmount: SMALLEST,
+    stakingDenom: network.stakingDenom,
+    network,
+    destinationValidatorAddress: null,
+    sort: {
+      property: `votingPower`,
+      order: `desc`,
+    },
+  }),
+  computed: {
+    ...mapState(`data`, [`delegations`]),
+    maximum() {
+      const delegation = this.delegations.find(
+        ({ validator }) =>
+          validator.operatorAddress === this.sourceValidator.operatorAddress
+      )
+      return delegation ? Number(delegation.amount) : 0
+    },
+    transactionData() {
+      return {
+        type: lunieMessageTypes.RESTAKE,
+        from:
+          this.sourceValidator && this.sourceValidator.operatorAddress
+            ? [this.sourceValidator.operatorAddress]
+            : null,
+        to: this.destinationValidatorAddress
+          ? [this.destinationValidatorAddress]
+          : null,
+        amount: {
+          amount: this.amount,
+          denom: this.stakingDenom,
+        },
+      }
+    },
+    notifyMessage() {
+      return {
+        title: `Successfully restaked!`,
+        body: `You have successfully restaked ${this.amount} ${this.stakingDenom}.`,
+      }
+    },
+    enhancedSourceValidator() {
+      return validatorEntry(this.sourceValidator)
+    },
+    sortedEnrichedValidators() {
+      const orderedValidators = orderBy(
+        this.validators,
+        [this.sort.property],
+        [this.sort.order]
```

Comment:
Can remove `sort` in data
```suggestion
        [`votingPower`],
        [`desc`]
```

Reply:
Fixed